code/game/objects/structures/mirror.dm

```diff
@@ -46,34 +40,31 @@
 
 		H.update_hair()
 
+		return TRUE
+
 /obj/structure/mirror/examine_status(mob/user)
 	if(broken)
 		return list()// no message spam
 	return ..()
 
-/obj/structure/mirror/obj_break(damage_flag, mapload)
+/obj/structure/mirror/obj_break(damage_flag)
 	if(!broken && !(flags_1 & NODECONSTRUCT_1))
-		icon_state = "mirror_broke"
-		if(!mapload)
-			playsound(src, "shatter", 70, 1)
+		icon_state = broken_icon_state
+		playsound(src, "shatter", 70, 1)
 		if(desc == initial(desc))
 			desc = "Oh no, seven years of bad luck!"
 		broken = TRUE
 
 /obj/structure/mirror/deconstruct(disassembled = TRUE)
 	if(!(flags_1 & NODECONSTRUCT_1))
 		if(!disassembled)
-			new /obj/item/shard( src.loc )
+			new /obj/item/shard(src.loc)
 	qdel(src)
 
 /obj/structure/mirror/welder_act(mob/living/user, obj/item/I)
 	if(user.a_intent == INTENT_HARM)
 		return FALSE
-
-	if(!broken)
-		return TRUE
-
-	if(!I.tool_start_check(user, amount=0))
+	if(!broken || !I.tool_start_check(user, amount=0))
 		return TRUE
 
 	to_chat(user, "<span class='notice'>You begin repairing [src]...</span>")

```

Comment:
this won't work for `/obj/structure/mirror/broken`

Reply:
Fixed, in a kinda-hacky-way